src/netbsd.rs

```diff
@@ -28,8 +25,14 @@ fn kern_arnd(buf: &mut [MaybeUninit<u8>]) -> libc::ssize_t {
 type GetRandomFn = unsafe extern "C" fn(*mut u8, libc::size_t, libc::c_uint) -> libc::ssize_t;
 
 pub fn getrandom_inner(dest: &mut [MaybeUninit<u8>]) -> Result<(), Error> {
+    fn link_getrandom() -> *mut c_void {
+        static NAME: &[u8] = b"getrandom\0";
+        unsafe { libc::dlsym(libc::RTLD_DEFAULT, NAME.as_ptr() as *const _) }
+    }
+
     // getrandom(2) was introduced in NetBSD 10.0
-    static GETRANDOM: Weak = unsafe { Weak::new("getrandom\0") };
+    static GETRANDOM: Weak = Weak::new(link_getrandom);
```

Comment:
It might be a little more readable to have these `static`s outside of `getrandom_inner`, like:

```rust
type GetRandomFn = unsafe extern "C" fn(*mut u8, libc::size_t, libc::c_uint) -> libc::ssize_t;
static NAME: &[u8] = b"getrandom\0";
static GETRANDOM: Weak = Weak::new(|| unsafe { libc::dlsym(libc::RTLD_DEFAULT, NAME.as_ptr() as *const _) });
```

having them all together makes the code a little easier to check.